### PR TITLE
CMake: Call function to setup the application target

### PR DESCRIPTION
**Pull request by hugueskamba**
The application target can only be configured from
where it has been created. Thus it needs to call the provided
function to apply settings required by Mbed OS.

Depends on: https://github.com/ARMmbed/mbed-os/pull/13413

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,8 @@ add_subdirectory(${MBED_ROOT})
 
 add_executable(${APP_TARGET})
 
+mbed_os_configure_app_target(${APP_TARGET})
```
Comment (0xc0170) [inline CMakeLists.txt:15]: we use `mbed_` prefix, not `mbed_os`, this should be fixed on CMake branch, not part of this PR - consistency

Comment (hugueskamba) [reply CMakeLists.txt:15]: Ah good point.
